### DISCUSSION TITLE
Ch.7 DeployFundMe.s.sol not deploying

### DISCUSSION THREAD

**Discussion by markcancode3000** [Cyfrin/foundry-full-course-cu#198] [Q&A] [answered]
I'm currently trying to deploy the DeployFundMe.s.sol contract in the terminal, however, every time I run (forge script script/DeployFundMe.s.sol) it brings up the following error. Error: contract source info format must be `<path>:<contractname>` or `<contractname>`. Haven't been able to figure out where I insert the `<path>:<contractname>` or `<contractname>` to fix the error. If anybody has suggestions please lmk. Thank you!
![Screen Shot 2023-06-30 at 12 03 52 AM](https://github.com/Cyfrin/foundry-full-course-f23/assets/95047407/a16580e7-7fc5-457c-b185-0f3f0a936157)

**Comment by alymurtazamemon** [comment, marked as answer]
@markcancode3000 You can use one of these two commands;
```bash
forge script script/DeployFundMe.s.sol:DeployFundMe
```
or
```bash
forge script DeployFundMe
```

The one you are using is the first one but you are missing the contract name with path separated by colon `:` `:DeployFundMe`

**Reply by markcancode3000** [reply]
Awesome! Thank you so much the second one worked perfectly!

**Reply by Bhavik-punmiya** [reply]
Thanks @alymurtazamemon  it helped me too the second solution to it. 
Can you give some reason why sometimes ``` forge script script/DeployFundMe.s.sol``` or any other contract deployment statement works but sometimes it does not, 
Still Thanks for these discussion I was trying to figure out on web but none helped !!

**Comment by ewebgh33** [comment]
Yeah I'm confused here, in the lesson when Patrick runs the script, (specifically for the SimpleStorage example) he doesn't add anything after `forge script ./script/DeploySimpleStorage.s.sol`
He does NOT type `forge script ./script/DeploySimpleStorage.s.sol:DeploySimpleStorage`

And so I don't understand why the command seems to be redundant. Run the script to deply... oh and also here's the name a 2nd time, just because, y'know.

Is this something new to Foundry since the tutorials were recorded?

**Comment by ewebgh33** [comment]
So I just tried
forge script ./script/DeploySimpleStorage.s.sol:DeploySimpleStorage
and I got back
```
Could not canonicalize the target path

Context:
- No such file or directory (os error 2)
```
And what kind of error message is this? Foundry is supposed to be so good but the errors are nonsensical. Where's the traceback? What target path? The one I just typed or the path to the script or the path to the contract IN the script?

**Comment by Jahunm** [comment]
guys can i get some help with this error?
foundry-f23/foundry-fund-me-f23$ forge script ./script/DeployFundMe.s.sol:DeployFundMe
[⠢] Compiling...
[⠊] Compiling 1 files with 0.8.25
[⠒] Solc 0.8.25 finished in 6.71s
Compiler run successful!
**Error: 
Could not find target contract**

**Reply by Jahunm** [reply]
thanks
